### ISSUE TITLE
[FEATURE]

### ISSUE THREAD

**Issue by SPPearce** [nf-core/modules#5848]
### Is your feature request related to a problem? Please describe

In the last few months we have swapped the file paths used in tests from:
file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
to direct paths like:
file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)

[This is in the guidelines now](https://nf-co.re/docs/guidelines/components/modules#input-data), but could do with a bulk update.

### Describe the solution you'd like

It would be great to have an automatic method to update all the filepaths at once, at least in the nf-test tests (but should be equivalent for pytest I think)
This could do with a linting check too, to ensure that new modules (or pytest -> nf-test conversions) still keep this.

**Comment by GallVp** [comment]
Hi @SPPearce 

I have started this in batches. The first PR is here: https://github.com/nf-core/modules/pull/6048

I have created a script to detect and then automatically fix the paths which is available elsewhere: 

1. https://github.com/GallVp/nxf-components/blob/main/.github/check_test_data_paths.sh
2. https://github.com/GallVp/nxf-components/blob/main/.github/get_test_data_path_update.nf

**Comment by SPPearce** [comment]
Oh thank you @GallVp !
Does it cope with paths specified with either single or double quotes?

**Comment by GallVp** [comment]
Yes, I think it does. It uses single quotes for the updated paths. I'll keep testing it and post updates here.

**Comment by GallVp** [comment]
> Oh thank you @GallVp ! Does it cope with paths specified with either single or double quotes?

If you mean params.test_data["xyz"], then the current version does not cope with that: https://github.com/GallVp/nxf-components/blob/6eaf5c8313ddd3d0672c3388cfe981bafb872d70/.github/get_test_data_path_update.nf#L18

Needs an update!

**Comment by SPPearce** [comment]
Yeah, not sure if that is generally used but worth checking. 
I have noticed some people put single or double quotes in the new file paths

**Comment by SPPearce** [comment]
I can't see any use of `params.test_data["xyz"]` with a quick search, so probably don't need to worry about that.

**Comment by GallVp** [comment]
Batch2: https://github.com/nf-core/modules/pull/6054

**Comment by GallVp** [comment]
Batch3: https://github.com/nf-core/modules/pull/6057

**Comment by GallVp** [comment]
Batch4: https://github.com/nf-core/modules/pull/6058

**Comment by GallVp** [comment]
Batch5: https://github.com/nf-core/modules/pull/6059

**Comment by GallVp** [comment]
List of modules with issues with their nf-tests

- `fastqscreen/fastqscreen` depends on `bowtie/build`; conda fail; https://github.com/nf-core/modules/issues/6075
- `centrifuge` depends on `untar`; permissions failure on `docker_self_hosted` runner; https://github.com/nf-core/modules/issues/5970
- `hamronization/rgi` triggers `tar` which fails
- There seems to be a new update for merfin: https://github.com/arangrhie/merfin which has to first go through Bioconda. For now, skipping `merfin/hist`.

**Comment by atrigila** [comment]
I think this one has been addressed and we can close it, could you confirm @SPPearce ?

**Comment by SPPearce** [comment]
Yep, the last few that had snuck in were completed in #10029
